File: packages/server/app/models/product.py
from typing import Literal

from pydantic import BaseModel


class Product(BaseModel):
    type: Literal["product"] = "product"
    ean: str
    name: str
    brand: str
    category: str
    price: float
    currency: str = "SEK"
    vat_rate: float
    image_url: str
    store_id: str
    in_stock: bool = True
    updated_at: str

File: packages/shared/src/models/product.ts
export interface Product {
  type: "product";
  ean: string;
  name: string;
  brand: string;
  category: string;
  price: number;
  currency: string;
  vat_rate: number;
  image_url: string;
  store_id: string;
  in_stock: boolean;
  updated_at: string;
}
